File: neetcode_solutions/arrays_and_hashing/two_sum/twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
const twoSum = function(nums, target) {
    
    differences = new Map();

    for (let i=0; i < nums.length; i++) {
        if (differences.has(target-nums[i])) return [i, differences.get(target-nums[i])];
        differences.set(nums[i], i)
    }
    return null;
};

File: blind_150_solutions/arrays and hashing/two_sum/two_sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """

        differences = {}
        for i, num in enumerate(nums):
            if target-num in differences:
                return [i, differences[target-num]]
            differences[num] = i
        return None
